contracts/services/generic-ibc-transfer/src/msg.rs

```diff
@@ -0,0 +1,286 @@
+    ) -> Self {
+        Self {
+            channel_id,
+            port_id,
+            ibc_transfer_timeout,
+        }
+    }
+}
+
+impl ServiceConfig {
+    pub fn new(
+        input_addr: ServiceAccountType,
+        output_addr: String,
+        denom: UncheckedDenom,
+        amount: IbcTransferAmount,
+        memo: String,
+        remote_chain_info: RemoteChainInfo,
+    ) -> Self {
+        Self {
+            input_addr,
+            output_addr,
+            denom,
+            amount,
+            memo,
+            remote_chain_info,
+            denom_to_pfm_map: BTreeMap::default(),
+        }
+    }
+
+    pub fn with_pfm_map(
+        input_addr: ServiceAccountType,
+        output_addr: String,
+        denom: UncheckedDenom,
+        amount: IbcTransferAmount,
+        memo: String,
+        remote_chain_info: RemoteChainInfo,
+        denom_to_pfm_map: BTreeMap<String, PacketForwardMiddlewareConfig>,
+    ) -> Self {
+        Self {
+            input_addr,
+            output_addr,
+            denom,
+            amount,
+            memo,
+            remote_chain_info,
+            denom_to_pfm_map,
+        }
+    }
+
+    fn do_validate(&self, api: &dyn cosmwasm_std::Api) -> Result<Addr, ServiceError> {
+        let input_addr = self.input_addr.to_addr(api)?;
+
+        match self.amount {
+            IbcTransferAmount::FullAmount => {}
+            IbcTransferAmount::FixedAmount(amount) => {
+                if amount.is_zero() {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid IBC transfer config: amount cannot be zero.".to_string(),
+                    ));
+                }
+            }
+        }
+
+        if self.remote_chain_info.channel_id.is_empty() {
+            return Err(ServiceError::ConfigurationError(
+                "Invalid IBC transfer config: remote_chain_info's channel_id cannot be empty."
+                    .to_string(),
+            ));
+        }
+
+        if let Some(port_id) = &self.remote_chain_info.port_id {
+            if port_id.is_empty() {
+                return Err(ServiceError::ConfigurationError(
+                    "Invalid IBC transfer config: remote_chain_info's port_id cannot be empty (if specified)."
+                        .to_string(),
+                ));
+            }
+        }
+
+        if let Some(timeout) = self.remote_chain_info.ibc_transfer_timeout {
+            if timeout.is_zero() {
+                return Err(ServiceError::ConfigurationError(
+                    "Invalid IBC transfer config: remote_chain_info's ibc_transfer_timeout cannot be zero.".to_string(),
+                ));
+            }
+        }
+
+        if !self.denom_to_pfm_map.is_empty() {
+            return Err(ServiceError::ConfigurationError(
+                "Invalid IBC transfer config: specifying a denom_to_pfm_map on the Generic IBC transfer service is currently unsupported.".to_string(),
+            ));
```

Comment:
Why? I think this needs to be supported on both (since logic would be shared no need to test on both services though)